### PR TITLE
Update to latest susbtrate and contracts pallet

### PR DESCRIPTION
**Pull request by ascjones**
Contracts `3.0` is here: https://github.com/paritytech/substrate/pull/8175.

This PR updates to substrate master (currently https://github.com/paritytech/substrate/commit/e6ac7e72c2251b8a23454c8cf0e3a2ad2c1ed566). For now we will continue to use github dependencies for `substrate` in aniticipation of migration to a parachain #30, which requires dependencies on `cumulus` and `polkadot`.

- The `Address` is now a `sp_runtime::MultiAddress<AccountId, ()>`, as per the node template (see https://github.com/paritytech/substrate/pull/7380), so clients need to be updated accordingly.
- Contracts pallet `parameter_types!` are copied verbatim from the substrate `node-runtime` https://github.com/paritytech/substrate/blob/36b8a962b254dc4d834477469a68c83c3060dd6b/bin/node/runtime/src/lib.rs#L716-L739

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -262,15 +264,41 @@ impl pallet_transaction_payment::Config for Runtime {
...
+// parameter_types! {
+// 	pub const TombstoneDeposit: Balance = 16 * MILLICENTS;
+// 	pub const RentByteFee: Balance = 100_000;
+// 	pub const RentDepositOffset: Balance = 1000 * MILLICENTS;
+// 	pub const SurchargeReward: Balance = 150 * MILLICENTS;
+// 	pub const SignedClaimHandicap: u32 = 2;
+// 	pub const MaxDepth: u32 = 32;
+// 	pub const StorageSizeOffset: u32 = 8;
+// 	pub const MaxValueSize: u32 = 16 * 1024;
+// }
```
Comment (ascjones) [inline runtime/src/lib.rs:276]: These are the existing contracts parameters, I have for now copied over the new parameters from the `node-runtime` :point_down: , so these probably need updating with some help from @athei

Comment (ascjones) [reply runtime/src/lib.rs]: Using values copied from substrate node runtime

**Review by athei — APPROVED** [review]
Cannot really comment on this client scaffold but the runtime looks good to me.

**Comment by ascjones** [comment]
> Cannot really comment on this client scaffold but the runtime looks good to me.

Client scaffold is just boilerplate copied from `node-template`

**Pull request merged**